streams/src/main/java/org/apache/kafka/streams/kstream/Produced.java

```diff
@@ -42,6 +42,12 @@ private Produced(final Serde<K> keySerde,
         this.partitioner = partitioner;
     }
 
+    public Produced(final Produced<K, V> produced) {
```

Comment:
Should this be protected? Compare the discussion in https://github.com/apache/kafka/pull/3807#discussion_r137603218